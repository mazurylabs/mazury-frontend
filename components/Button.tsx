import React from 'react';

interface ButtonProps {
  children: React.ReactNode;
  onClick?: () => void;
  className?: string;
  disabled?: boolean;
}

export const Button: React.FC<ButtonProps> = ({
  children,
  onClick,
  className = '',
  disabled = false,
}) => {
  return (
    <button
      onClick={onClick}
      className={`bg-slate-200 px-4 py-1 rounded ${className} disabled:opacity-50`}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

export const OutlineButton: React.FC<ButtonProps> = ({
  children,
  className,
  ...props
}) => {
  return (
    <Button
      className={`border-2 border-blue-800 text-blue-800 shadow rounded-lg bg-white px-8 flex gap-2 items-center ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};
