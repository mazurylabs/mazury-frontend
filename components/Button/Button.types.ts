export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  variant?: 'primary' | 'secondary' | 'tertiary';
  className?: string;
  size?: 'small' | 'large';
  loading?: boolean;
}
