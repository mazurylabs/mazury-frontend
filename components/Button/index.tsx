export { Button, BlueSocialButton } from './Button';
