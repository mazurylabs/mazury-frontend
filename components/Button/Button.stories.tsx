import { ComponentMeta, ComponentStory } from '@storybook/react';
import * as React from 'react';
import { HiSearch } from 'react-icons/hi';

import { Button } from './Button';

export default {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    children: {
      control: { type: 'text' },
      disabled: { type: 'boolean' },
    },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: 'BUTTON',
};

export const Secondary = Template.bind({});
Secondary.args = {
  children: 'BUTTON',
  variant: 'secondary',
};

export const Tertiary = Template.bind({});
Tertiary.args = {
  children: 'BUTTON',
  variant: 'tertiary',
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  children: (
    <>
      <HiSearch />
      <span>BUTTON</span>
    </>
  ),
};
