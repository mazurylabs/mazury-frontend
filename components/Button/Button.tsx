import * as React from 'react';
import { Spinner } from '../Spinner';
import { ButtonProps } from './Button.types';

const variants = {
  primary:
    'bg-indigoGray-90  text-indigoGray-5 active:bg-indigoGray-70 active:text-indigoGray-10  disabled:bg-indigoGray-20 shadow-base hover:bg-indigoGray-50 hover:text-indigoGray-20 disabled:text-indigoGray-40',
  secondary:
    'border-[1.5px] border-indigoGray-20 bg-indigoGray-10 text-indigoGray-90 hover:border-indigoGray-30 hover:text-indigoGray-50 active:border-indigoGray-80 active:text-indigoGray-80 disabled:border-indigoGray-30 disabled:bg-indigoGray-20 disabled:text-indigoGray-40 hover:bg-indigoGray-20 hover:text-indigoGray-50 active:bg-indigoGray-5 active:border-indigoGray-20 active:text-indigoGray-70',
  tertiary:
    'bg-none text-indigoGray-90 shadow-none hover:text-indigoGray-50 active:text-indigoGray-70 disabled:text-indigoGray-40',
};

export const Button: React.FC<ButtonProps> = ({
  children,
  variant = 'primary',
  className,
  size = 'small',
  loading,
  type = 'button',
  ...rest
}) => {
  return (
    <button
      className={`flex items-center justify-center gap-2 rounded-[8px] ${
        size === 'small' ? 'py-2' : 'py-3'
      } px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none ${
        variants[variant]
      } ${className}`}
      {...rest}
      type={type}
    >
      <>{children}</>
      {loading && (
        <Spinner
          size={16}
          className="ml-2"
          variant={variant === 'primary' ? 'light' : 'dark'}
        />
      )}
    </button>
  );
};

export const BlueSocialButton: React.FC<ButtonProps> = ({
  children,
  onClick,
  className,
}) => {
  return (
    <button
      onClick={onClick}
      className={`flex items-center gap-2 rounded-xl border-2 border-blue-800 py-1 px-4 text-blue-800 shadow-base disabled:cursor-not-allowed disabled:shadow-none ${className}`}
    >
      {children}
    </button>
  );
};
