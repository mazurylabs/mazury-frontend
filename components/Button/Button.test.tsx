import { render, screen } from '@testing-library/react';
import { FaSearch } from 'react-icons/fa';
import { Button } from './Button';

describe('Button', () => {
  test('renders a button', () => {
    render(<Button>Button</Button>);
    const btn = screen.getByRole('button', {
      name: 'Button',
    });
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('secondary variant', () => {
    render(<Button variant="primary">Button</Button>);
    const btn = screen.getByRole('button', {
      name: 'Button',
    });
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('tertiary variant', () => {
    render(<Button variant="tertiary">Button</Button>);
    const btn = screen.getByRole('button', {
      name: 'Button',
    });
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('multiple children (SVG icon)', () => {
    render(
      <Button variant="tertiary">
        <FaSearch data-testid="icon" /> <span>Search</span>
      </Button>
    );
    const btn = screen.getByRole('button', {
      name: 'Search',
    });
    const icon = screen.getByTestId('icon');
    expect(btn).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });

  test('disabled state', () => {
    render(<Button disabled>Button</Button>);
    const btn = screen.getByRole('button', {
      name: 'Button',
    });
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
  });
});
