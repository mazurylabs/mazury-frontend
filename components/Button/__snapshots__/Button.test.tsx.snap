// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled state 1`] = `
<button
  class="flex items-center justify-center gap-2 rounded-[8px] py-2 px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none bg-indigoGray-90  text-indigoGray-5 active:bg-indigoGray-70 active:text-indigoGray-10  disabled:bg-indigoGray-20 shadow-base hover:bg-indigoGray-50 hover:text-indigoGray-20 disabled:text-indigoGray-40 undefined"
  disabled=""
  type="button"
>
  Button
</button>
`;

exports[`Button multiple children (SVG icon) 1`] = `
<button
  class="flex items-center justify-center gap-2 rounded-[8px] py-2 px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none bg-none text-indigoGray-90 shadow-none hover:text-indigoGray-50 active:text-indigoGray-70 disabled:text-indigoGray-40 undefined"
  type="button"
>
  <svg
    data-testid="icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
    />
  </svg>
   
  <span>
    Search
  </span>
</button>
`;

exports[`Button renders a button 1`] = `
<button
  class="flex items-center justify-center gap-2 rounded-[8px] py-2 px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none bg-indigoGray-90  text-indigoGray-5 active:bg-indigoGray-70 active:text-indigoGray-10  disabled:bg-indigoGray-20 shadow-base hover:bg-indigoGray-50 hover:text-indigoGray-20 disabled:text-indigoGray-40 undefined"
  type="button"
>
  Button
</button>
`;

exports[`Button secondary variant 1`] = `
<button
  class="flex items-center justify-center gap-2 rounded-[8px] py-2 px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none bg-indigoGray-90  text-indigoGray-5 active:bg-indigoGray-70 active:text-indigoGray-10  disabled:bg-indigoGray-20 shadow-base hover:bg-indigoGray-50 hover:text-indigoGray-20 disabled:text-indigoGray-40 undefined"
  type="button"
>
  Button
</button>
`;

exports[`Button size prop works as intended 1`] = `
<button
  class="flex items-center justify-center gap-2 rounded-[8px] py-3 px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none bg-indigoGray-90  text-indigoGray-5 active:bg-indigoGray-70 active:text-indigoGray-10  disabled:bg-indigoGray-20 shadow-base hover:bg-indigoGray-50 hover:text-indigoGray-20 disabled:text-indigoGray-40 undefined"
  type="button"
>
  Button
</button>
`;

exports[`Button tertiary variant 1`] = `
<button
  class="flex items-center justify-center gap-2 rounded-[8px] py-2 px-6 font-sans text-sm font-semibold disabled:cursor-not-allowed disabled:shadow-none bg-none text-indigoGray-90 shadow-none hover:text-indigoGray-50 active:text-indigoGray-70 disabled:text-indigoGray-40 undefined"
  type="button"
>
  Button
</button>
`;
